python/lsst/pipe/base/mermaid_tools.py

```diff
@@ -0,0 +1,431 @@
+    print("```mermaid", file=file)
+    print("flowchart TD", file=file)
+
+    # To avoid duplicating dataset nodes, we track them.
+    allDatasetRefs: dict[str, str] = {}
+
+    # Process each task/quantum.
+    for taskId, taskDef in enumerate(qgraph.taskGraph):
+        quanta = qgraph.getNodesForTask(taskDef)
+        for qId, quantumNode in enumerate(quanta):
+            # Create quantum node.
+            taskNodeName = f"TASK_{taskId}_{qId}"
+            taskLabelLines = [f"**{taskDef.label}**", f"Node ID: {quantumNode.nodeId}"]
+            dataId = quantumNode.quantum.dataId
+            if dataId is not None:
+                for k in sorted(dataId.required.keys()):
+                    taskLabelLines.append(f"{k}={dataId[k]}")
+            else:
+                raise ValueError("Quantum DataId cannot be None")
+            taskLabel = "<br>".join(taskLabelLines)
+            print(f'{taskNodeName}["{taskLabel}"]', file=file)
+
+            # Quantum inputs: datasets --> tasks
+            for dsRefs in quantumNode.quantum.inputs.values():
+                for dsRef in dsRefs:
+                    dsNode = _makeDatasetNode(dsRef, allDatasetRefs, file)
+                    print(f"{dsNode} --> {taskNodeName}", file=file)
+
+            # Quantum outputs: tasks --> datasets
+            for dsRefs in quantumNode.quantum.outputs.values():
+                for dsRef in dsRefs:
+                    dsNode = _makeDatasetNode(dsRef, allDatasetRefs, file)
+                    print(f"{taskNodeName} --> {dsNode}", file=file)
+
+    # End Mermaid code block.
+    print("```", file=file)
+
+    if close:
+        file.close()
+
+
+def _expand_dimensions(dimension_list: list[str], universe: DimensionUniverse) -> list[str]:
+    """Return expanded list of dimensions, with special skypix treatment.
+
+    Parameters
+    ----------
+    dimension_set : `list` [`str`]
+        The original set of dimension names.
+    universe : DimensionUniverse
+        Used to conform the dimension set according to a known schema.
+
+    Returns
+    -------
+    dimensions : `list` [`str`]
+        Expanded list of dimensions.
+    """
+    dimension_set = set(dimension_list)
+    skypix_dim = []
+    if "skypix" in dimension_set:
+        dimension_set.remove("skypix")
+        skypix_dim = ["skypix"]
+    dimensions = universe.conform(dimension_set)
+    return list(dimensions.names) + skypix_dim
+
+
+def _format_dimensions(dims: list[str]) -> str:
+    """Format and sort dimension names as a comma-separated list inside curly
+    braces.
+
+    For example, if dims=["detector", "visit"], returns "{detector,visit}".
+
+    Parameters
+    ----------
+    dims : list of str
+        The dimension names to format and sort.
+
+    Returns
+    -------
+    str
+        The formatted dimension string, or an empty string if no dimensions.
+    """
+    if not dims:
+        return ""
+    sorted_dims = sorted(dims)
+    return "{" + ", ".join(sorted_dims) + "}"
```

Comment:
If you changed this to:
```
return "{" + ",&nbsp;".join(sorted_dims) + "}"
```
I think it would prevent line breaks across dimensions, which would aid in readability and more closely mimic the existing DOT outputs.